Libraries/ConnectNIO/Internal/ConnectUnaryChannelHandler.swift

```diff
@@ -51,6 +52,7 @@ final class ConnectUnaryChannelHandler: NIOCore.ChannelInboundHandler, @unchecke
             }
 
             self.closeConnection()
+            self.hasResponded = true
```

Comment:
Alternatively to my previous comment, what do you think about moving `self.hasResponded = true` to `closeConnection`? Seems like that would simplify this a bit